pkg/resources/digitalocean/dns/dns.go

```diff
@@ -317,91 +318,42 @@ func (r *resourceRecordChangeset) Apply() error {
 		return nil
 	}
 
-	if len(r.additions) > 0 {
-		for _, record := range r.additions {
-			recordCreateRequest := &godo.DomainRecordEditRequest{
-				Name: record.Name(),
-				Data: record.Rrdatas()[0],
-				TTL:  int(record.Ttl()),
-				Type: string(record.Type()),
-			}
-			err := createRecord(r.client, r.zone.Name(), recordCreateRequest)
-			if err != nil {
-				return fmt.Errorf("could not create record: %v", err)
-			}
-		}
-
-		glog.V(2).Infof("record change set additions complete")
-	}
-
 	if len(r.removals) > 0 {
```

Comment:
This could be problematic, if we temporarily delete a DNS record that is important, and there's a delay before we recreate it.

Hopefully everything is using upserts instead of add / remove pairs, but the distinction hasn't mattered as much where we've had batch operations.

We inherited this API from the federation code, and I added `upsert` IIRC.  We could also look at how `external-dns` is handling it - the long term goal is to get closer to that code :-)

Reply:
Agreed, I don't love this implementation and it seems error prone, but DNS API is not setup in a way that plays nicely with the interface from the federation code. I'll take a look at external-dns and let you know if something better comes up :)